src/preact/base-element.js

```diff
@@ -400,7 +400,7 @@ export class PreactBaseElement extends BaseElement {
     const {staticProps} = Ctor;
     const initProps = this.init();
     Object.assign(
-      /** @type {!Object} */ (this.defaultProps_),
+      /** @type {Object} */ (this.defaultProps_),
```

Comment:
This cast may not be necessary, since in theory almost everything inherits from `Object` and `this.defaultProps_` is a `JsonObject`. Then again, this file isn't even type-checked, so ¯\\_(ツ)_/¯

Reply:
This PR adds type checking!

Reply:
forgot to test this out, will do before merge

Reply:
Yep, this was completely unecessary (it was preexisting. during this PR I hadn't thought to remove unecessary pre-existing casts!)